include/oneapi/dpl/pstl/hetero/dpcpp/sycl_iterator.h

```diff
@@ -118,6 +118,12 @@ struct sycl_iterator
     {
         return buffer;
     }
+
+    Size
+    get_idx() const
```

Comment:
Do I understand correctly that "soclomatic"  uses such API?

Reply:
SYCLomatic does not use `sycl_iterator` at all, but rather creates a type similar to `sycl_iterator`  (`device_pointer` / `device_iterator`) which are similar.  They already provide `get_idx()` within those types.  

The idea is to require these `is_hetero` types to provide `get_idx()` and then use it within `__get_sycl_range` processing.

Reply:
Don't quite understand. `__get_sycl_range `works fine without `get_idx()`.

Reply:
... Actually, the answer the next comment  explain that. I see..